contracts/src/locker/token_locker.cairo

```diff
@@ -312,8 +325,7 @@ mod TokenLocker {
         /// Internally, this function reads the list of locks of the specified `owner` from the `user_locks` mapping.
         /// It then iterates over the list and replaces the specified `lock_id` with the last element of the list.
         /// The length of the list is then decremented by one, and the last element of the list is set to zero.
-        fn remove_user_lock(self: @ContractState, lock_id: u128, owner: ContractAddress) {
-            let mut list = self.user_locks.read(owner);
+        fn remove_user_lock(self: @ContractState, lock_id: u128, mut list: List<u128>) {
```

Comment:
I feel arguments list should be more descriptive like: `user_locks`

```suggestion
        fn remove_user_lock(self: @ContractState, lock_id: u128, mut user_locks: List<u128>) {
```

Reply:
Also, ownership of list (user_locks) will be given to this function making it unusable in caller function. Is this expected ?

Reply:
> I feel arguments list should be more descriptive like: user_locks

The purpose of this PR was to refactor this function to remove an element from _any_ list, not just user locks. Here the part to adapt is more the function name and description that the name of the list passed, which can be anything.

Reply:
> Also, ownership of list (user_locks) will be given to this function making it unusable in caller function. Is this expected ?

I don't think we ever need to use the list after removing items from it - which only happens upon unlocking a token - so I would say this is the right move